### PR TITLE
broken polls can now be fixed or removed from voting

### PR DESCRIPTION
**Pull request by Intigracy**
gonna need someone versed in SQL to look this over, which afaik is just @d3athrow

### CODE REVIEW

File: code/modules/admin/polls/fix_poll.dm [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+			log_game("SQL ERROR adding new poll option to table. Error : \[[err]\]\n")
+			return
+		else
+			i++
+		switch(alert("Add more options?",,"Add option","Finish"))
+			if("Add option")
+				add_option = 1
+			if("Finish")
+				add_option = 0
+				return
```
Comment (ghost) [inline code/modules/admin/polls/fix_poll.dm]: Trailing

Comment (ghost) [reply code/modules/admin/polls/fix_poll.dm]: Wait, it's not even trailing, it's a while loop exit. Disgusting.

Comment (ghost) [reply code/modules/admin/polls/fix_poll.dm]: And the worst thing is that the line before prevents the while loop continuing anyway.

File: code/modules/admin/polls/fix_poll.dm [hunk 2]
```diff
@@ -0,0 +1,91 @@
+/client/proc/fix_poll()
+	set name = "Fix Poll"
+	set category = "Special Verbs"
+	if(!check_rights(R_POLLING))
+		return
+	if(!dbcon.IsConnected())
+		src << "<span class='danger'>Failed to establish database connection.</span>"
```
Comment (PJB3005) [inline code/modules/admin/polls/fix_poll.dm]: `to_chat`

File: code/modules/admin/polls/fix_poll.dm [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+		var/descmax = ""
+		if(polltype == "NUMVAL")
+			minval = input("Set minimum rating value.","Minimum rating") as num
+			if(!minval)
+				return
+			maxval = input("Set maximum rating value.","Maximum rating") as num
+			if(!maxval)
+				return
+			if(minval >= maxval)
+				src << "Minimum rating value can't be more than maximum rating value"
```
Comment (PJB3005) [inline code/modules/admin/polls/fix_poll.dm]: Hrm

File: code/modules/admin/polls/fix_poll.dm [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+	var/selectedpoll = input("Select poll lacking options that you'd like to fix.","THEY'RE BROKEN") as null|anything in broken_polls
+	if(!selectedpoll)
+		return
+
+	var/dowhat = alert(src, "What would you like to do with this poll? Please note that if this poll's question matches another one verbatum, this will affect both.","What do you plan to do?","End Poll","Add Options","Cancel")
+	if(dowhat == "Cancel")
+		return
+
+	if(dowhat == "End Poll") //Verbatum matching questions are all affected by this.
+		var/DBQuery/end_this_poll = dbcon.NewQuery("UPDATE erro_poll_question SET endtime = NOW() WHERE question = '[selectedpoll]' AND WHERE option IS NULL AND polltype != 'TEXT' AND DATE(endtime) > NOW()")
```
Comment (ComicIronic) [inline code/modules/admin/polls/fix_poll.dm:27]: Every day I thank the lord for Byond's thin client.

**Comment by Intigracy** [comment]
Just realized this needs fixing, option doesn't exist on erro_poll_question

**Comment by clusterfack** [comment]
So fix it?

**Comment by Intigracy** [comment]
Naw, really?

**Comment by clusterfack** [comment]
Fix it?

**Comment by clusterfack** [comment]
Fix

**Comment by Intigracy** [comment]
Yes
I had work today

**Comment by Intigracy** [comment]
Sunday

**Comment by Intigracy** [comment]
what a bully

**Comment by clusterfack** [comment]
Wee woo wee woo

**Comment by clusterfack** [comment]
Okay well just, reopen when its finished then I guess

**Comment by Intigracy** [comment]
oh nooooooooooooooo

**Pull request closed without merging**